Sources/StreamChat/Workers/UserUpdater.swift

```diff
@@ -27,6 +27,60 @@ class UserUpdater: Worker {
             completion?($0.error)
         }
     }
+    
+    /// Blocks the user with the provided `userId`.
+    /// - Parameters:
+    ///   - userId: The user identifier.
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    ///
+    func blockUser(_ userId: UserId, completion: ((Error?) -> Void)? = nil) {
+        apiClient.request(endpoint: .blockUser(userId)) {
+            switch $0 {
+            case .success:
+                self.database.write({ session in
+                    session.currentUser?.blockedUserIds.insert(userId)
+                }, completion: {
+                    if let error = $0 {
+                        log.error("Failed to save blocked user with id: <\(userId)> to the database. Error: \(error)")
+                        
+                        self.database.write({ session in
+                            session.currentUser?.blockedUserIds.remove(userId)
+                        })
+                    }
+                    completion?($0)
+                })
+            case let .failure(error):
+                completion?(error)
+            }
```

Comment:
The remove of the userId, should be done in the `failure` case, which is where the http request fails. Currently is being done when the DB write fails.

Reply:
Updated 👍